EpikCord/__init__.py

```diff
@@ -1156,6 +1159,23 @@ async def command_error(
     ):
         logger.exception(error)
 
+    async def auto_moderation_rule_create(self, data:dict) -> AutoModerationRule:
+        rule = AutoModerationRule(self, data)
+        self.auto_moderation_cache.add_to_cache(rule.guild_id, rule)
+        return rule
+    
+    async def auto_moderation_rule_update(self, data:dict)-> tuple[Optional[AutoModerationRule], AutoModerationRule]:
+        after:AutoModerationRule = AutoModerationRule(self,data)
+        before:Optional[AutoModerationRule] = self.auto_moderation_cache.get(after.guild_id)
+        return before,after
+        
+    async def auto_moderation_rule_delete(self, data:dict)->Optional[AutoModerationRule]:
+        #According to discord docs, they will send an automoderation object , but i think it will be blank rule
+        # In this case, we need a previous instance, so discord's reply is not feasible
+        # Until the docs are clarified, I think our best bet here is to take the previous one from the cache
```

Comment:
```suggestion
```

Reply:
This shouldn't be a comment in the package but more of an implementation open question to the contributors & core team